modules/sensor/src/rgb-depth/realsense/vpRealSense2.cpp

```diff
@@ -641,9 +641,10 @@ void vpRealSense2::acquire(unsigned char *const data_image, unsigned char *const
  */
 void vpRealSense2::close()
 {
-  if (m_init) {
-    m_pipe.stop();
-    m_init= false;
+  if (m_pipe != NULL) {
+    m_pipe->stop();
+    delete m_pipe;
+    delete m_pipelineProfile;
   }
```

Comment:
here you should add `m_pipe = NULL` and `m_pipelineProfile = NULL` after the delete for security